game_event/tests.py

```diff
@@ -265,3 +296,53 @@ def test_fail_add_player_to_event_already_in_it(self, client, saved_game_event,
         assert response.status_code == 200
         with pytest.raises(IntegrityError):
             response = client.post(url, {'answer': 'yes'})
+
+
+@pytest.mark.django_db
+class TestWeatherApi:
+    def test_one_valid_response(self, client, saved_game_event, player, mock_urlopen):
+        mock_response = mock.Mock()
+        mock_response.status = 200
+        mock_response.read.return_value = bytes('{"weather": [{"icon": "04d"}],"main": {"temp": 13.19}}', 'utf-8')
+        mock_urlopen.return_value = mock_response
+
+        client.force_login(player.user)
+
+        output = get_weather_from_api([saved_game_event], mock_urlopen)
+        assert output[saved_game_event.id] == {'icon': "http://openweathermap.org/img/wn/04d@2x.png", 'temp': 13.19}
+
+    def test_exception_thrown_from_api_call(self, client, saved_game_event, player, mock_urlopen):
+
+        mock_urlopen.side_effect = Exception()
+        client.force_login(player.user)
+
+        output = get_weather_from_api([saved_game_event], mock_urlopen)
+        assert output[saved_game_event.id] is None
+
+    def test_one_invalid_response_status_code(self, client, saved_game_event, player, mock_urlopen):
+        mock_response = mock.Mock()
+        mock_response.status = 185
+        mock_urlopen.return_value = mock_response
+
+        client.force_login(player.user)
+
+        output = get_weather_from_api([saved_game_event], mock_urlopen)
+        assert output[saved_game_event.id] is None
+
+    def test_empty_game_events_list_valid_empty_dictionary_response(self, client, player, mock_urlopen):
+        client.force_login(player.user)
+        output = get_weather_from_api([], "does not matter")
+        assert output == {}
+
+    def test_two_valid_game_events_list_valid_response(self, client, saved_game_events_list, player, mock_urlopen):
+        mock_response = mock.Mock()
+        mock_response.status = 200
+        mock_response.read.return_value = bytes('{"weather": [{"icon": "04d"}],"main": {"temp": 13.19}}', 'utf-8')
+        mock_urlopen.return_value = mock_response
+
+        client.force_login(player.user)
+
+        for index in range(len(saved_game_events_list)):
+            output = get_weather_from_api([saved_game_events_list[index]], mock_urlopen)
+            assert output[saved_game_events_list[index].id] == {'icon': "http://openweathermap.org/img/wn/04d@2x.png",
+                                                                'temp': 13.19}
```

Comment:
This test is not testing 2 entries. It tests calling the function twice with a different entry.

I'd suggest that instead of having a fixture for the list of events, you add another fixture for the second event and construct the list here. e.g. `saved_game_event, another_saved_game_event`
```suggestion
        output = get_weather_from_api([saved_game_event, another_saved_game_event], mock_urlopen)
        expected_weather = {'icon': "http://openweathermap.org/img/wn/04d@2x.png", 'temp': 13.19}
        assert output == {
                saved_game_event.id: expected_weather,
                another_saved_game_event: expected_weather,
        }
        for index in range(len(saved_game_events_list)):
            output = get_weather_from_api([saved_game_events_list[index]], mock_urlopen)
            assert output[saved_game_events_list[index].id] == {'icon': "http://openweathermap.org/img/wn/04d@2x.png",
                                                                'temp': 13.19}
```